### ISSUE TITLE
Better parameters description?

### ISSUE THREAD

**Issue by derei** [alicevision/meshroom#710]
When trying to get my head around Meshroom, I started following tutorials and the Meshroom manual. Turns out, there isn't a lot of detailed explanation what each node does and especially HOW it does it. Even the tooltips in the program just repeat the name of the parameter in just more words.

It is clear that for new users it should be a clear description of each parameter for the nodes with what that parameter does to the process and how exactly it will affect the output if changed.

Right now, I'm trying to refine a mesh using `MeshFiltering, MeshDenoising, MeshResampling, MeshDecimate`... but the ambiguity is high.

Specific questions:
What is **Lambda**? - what does it do and what would happen specifically when altering its value? I've seen lambda as parameter in several nodes.
**MeshDenoising**:  _Eta, Mu, Nu_? - in layman's words, what this parameters do and given a desired result, how to change them? For example I want my mesh to keep the same shape, but to remove small surface noise?

Thank you, and let's all hope for a better user manual in the near future.

**Comment by natowi** [comment]
You can find important parameters here https://github.com/alicevision/meshroom/wiki/Reconstruction-parameters

There are many "advanced" parameters in the node settings to fine tune the used algorithms.

[MeshDenoise](https://meshroom-manual.readthedocs.io/en/latest/node-reference/nodes/MeshDenoising.html) 

Example:

Eta | Gaussian standard deviation for spatial weight, scaled by the  average distance between adjacent face centroids. Must be positive.(0.0 -  20) (1.5)

So some knowledge of the algorithms is required to effectively use these parameters. 

We are working on the [documentation](https://meshroom-manual.readthedocs.io/en/latest/) but it is not complete yet.

**Comment by derei** [comment]
Thank you for the explanation. I understand the people behind this project are highly technical and may not be so aware that most of us, the users may not have so much background information about this algorithms.

For example: _"Eta | Gaussian standard deviation for spatial weight, scaled by the average distance between adjacent face centroids. Must be positive.(0.0 - 20) (1.5)"_ is a completely abstract explanation for me. Although I understand the words in the phrase, for me they make no sense. A more approachable explanation is required, maybe accompanied by an example or a small graph showing how the mesh will be affected.

If such explanation is not provided, the users will have no choice but to randomly change parameters and see what happens. This can be very unpredictable and also very time consuming. Eventually will lead to frustration.

Yes, I tried to use the manual and the result was more confusion, as the description of the parameters is - as I said - too abstract and does not explain what will happen (an acceptable way of explaining it would be: higher value, will make the mesh smoother, but less accurate, as in this image <and show an example image with before/after>).

**Comment by natowi** [comment]
> A more approachable explanation is required, maybe accompanied by an example or a small graph showing how the mesh will be affected.

Yes, that is something being worked on, but the progress is slow, as Meshroom contributors are thinly spread.

**Comment by fabiencastan** [comment]
@derei The MeshDenoise node is an experimental node included to allow us to experiment with it. I fully agree that the parameters of this node are too difficult to manipulate to be usable. It's not a problem of documentation. The problem is that it's not explain anywhere that this node is experimental and not ready for end users. We should maybe just remove it from the releases.

Same regarding the "advanced parameters", they are useful for researchers and contributors to analyse in fine detail the impact of internal constants, variants of algorithms, etc. but they are not intended to be used by graphic artists. That's why we have made them hidden by default.

But I agree that more globally there is still a lack of documentation on the user parameters.

**Comment by derei** [comment]
In fact, I believe the MeshDenoise should be left available, but maybe added a description that is experimental (so people won't have unrealistic expectations). I really found it useful, although the results were obtained by trial and error.
Knowing what the parameters affect, maybr would save some time.

Thanks for clarifying this for me.
-- 
 Securely sent with Tutanota. Get your own encrypted, ad-free mailbox: 
 https://tutanota.com


24 Nov 2019, 14:47 by notifications@github.com:

**Comment by natowi** [comment]
@fabiencastan I would also keep the parameters and nodes in the release, but maybe move all dev. parameters to advanced. We could add tags to the node description chapter in the docs like a "ready" / "experimental" badge to clearly mark the state of the nodes in the documentation.
https://img.shields.io/badge/state-ready-brightgreen
https://img.shields.io/badge/state-experimental-blue

**Comment by curtthemartian** [comment]
I propose that if we don't have the item documented in a manual or wiki, and assuming we have access to a paper or github source be provided as a URL on the node selection.  It would be a quicker fix and more people might be able to help add documentation to the meshroom manual.   |
Of course if such a thing was there, it would have to be done carefully so as to not block a users interest in moving a slider.   Maybe have it as a right click menu only visible in the node window under the reset to default value button.

**Comment by tomhol** [comment]
@derei I feel your pain while reading your comments here. I'm in the same situation. I intuitively feel like the MeshDenoising node would improve the scan significantly. Just it is very frustrating to not have the parameters documented in simple words. I mean reasonably technical words but not "too scientific". The default values for me produce a denoised model, but just too denoised. Lot of features just diappeared. I guess this might relate to the fact that the scan contains the object itself (which is about 30 cm big) but also surroundings which are 2-3 meters away. I feel like adjusting one parameter or two would produce what I want just it is really painful to figure it by try-and-miss technique :|
I think there should be some basic documentation for each parameter, with examples. Like: parameter X - you wanna decrease it if the result is....; you wanna increase it when you experience ...; Some typical scenarios and typical parameter values. It is clear that the default values cannot suit all situations. Hence few examples and typical parameter sets will be handy.
BTW - @derei, did you figure some guidance yourself? Can you share in layman words here? I think this would be useful for many folks.
Thanks!

**Comment by natowi** [comment]
@tomhol You can read the paper I mentioned before https://arxiv.org/pdf/1712.03574.pdf
and you will find information like this: "A larger value of Lambda or Eta leads to a smoother filtering result." (Fig. 12)
You are welcome to extract the relevant information from the paper and I will add it to the documentation.

**Comment by derei** [comment]
@tomhol I am pleased to see other people having similar concerns, this gives hope that a better documentation will eventually emerge.
As for myself, I did not do any more progress unfortunately. It was for a uni project and I have more projects to work on, so anything on hobby side has been halted for now.

**Comment by polygonfuture** [comment]
I concur on all that has been said.    The current documentation for many nodes only explain things for those deeply involved in the development of the application itself or directly relating to another paper.   It is of no use to those who are not as intimately involved.   

Glad to see @natowi bringing up and mentioning the MeshSDFilter and the conerning paper.

**Comment by jacendvs** [comment]
While reading the bit about comparing a pixel to the average of its neighboring pixels, I wondered if an optical analysis using the depth of field resulting from different aperture sizes / dynamic aperture  / dynamic focal length could be useful in analyzing structure/features of the object being scanned.
